### PR TITLE
refactor: flex and style refactor attributes

### PR DESCRIPTION
**Pull request by diogooliveira-zup**
refactor: Flex and some attributes moved to Style to improve the responsabilities #101

The Style has new attributes that was in the Flex and Flex is part of Styles now, so if you need to set values to this attributes as (size, margin etc) refeer to Style instead of Flex
backgroundColor
cornerRadius
size
margin
padding
position
flex

BREAKING CHANGE: attributes backgroundColor, cornerRadius, size, margin,  padding, positionType, Display and Flex moved to Style class

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleFlexView.kt [hunk 1]
```diff
@@ -19,29 +19,26 @@ package br.com.zup.beagle.android.view
...
 import android.content.Context
 import android.view.View
-import br.com.zup.beagle.core.FlexComponent
-import br.com.zup.beagle.core.GhostComponent
-import br.com.zup.beagle.core.ServerDrivenComponent
 import br.com.zup.beagle.android.engine.mapper.FlexMapper
 import br.com.zup.beagle.android.engine.renderer.ViewRendererFactory
 import br.com.zup.beagle.android.widget.RootView
-import br.com.zup.beagle.widget.core.Flex
+import br.com.zup.beagle.core.*
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleFlexView.kt:25]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/20f1aa27594959ec0782fdc4d4c65b000d430d17/android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleFlexView.kt#L25"></span>br.com.zup.beagle.core.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/Widget.kt [hunk 2]
```diff
@@ -16,24 +16,16 @@
...
 package br.com.zup.beagle.widget
 
-import br.com.zup.beagle.core.Accessibility
-import br.com.zup.beagle.core.AccessibilityComponent
-import br.com.zup.beagle.core.FlexComponent
-import br.com.zup.beagle.core.IdentifierComponent
-import br.com.zup.beagle.core.Style
-import br.com.zup.beagle.core.StyleComponent
-import br.com.zup.beagle.widget.core.Flex
+import br.com.zup.beagle.core.*
```
Comment (botbeaglezup) [inline schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/Widget.kt:19]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/20f1aa27594959ec0782fdc4d4c65b000d430d17/schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/Widget.kt#L19"></span>br.com.zup.beagle.core.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleFlexView.kt (hunk 1 above)
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleFlexView.kt:25]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/20f1aa27594959ec0782fdc4d4c65b000d430d17/android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleFlexView.kt#L25"></span>br.com.zup.beagle.core.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/Widget.kt (hunk 2 above)
Comment (botbeaglezup) [inline schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/Widget.kt:19]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/20f1aa27594959ec0782fdc4d4c65b000d430d17/schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/widget/Widget.kt#L19"></span>br.com.zup.beagle.core.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/Spacer.kt [hunk 3]
```diff
@@ -33,13 +33,13 @@ data class Spacer(
     private val viewFactory: ViewFactory = ViewFactory()
 
     override fun buildView(rootView: RootView): View {
```
Comment (jefersonlopeszup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/Spacer.kt:35]:
```suggestion
   override fun buildView(rootView: RootView): View = viewFactory.makeBeagleFlexView(rootView.getContext(), Flex(
        size = Size(
            width = UnitValue(size, UnitType.REAL),
            height = UnitValue(size, UnitType.REAL)
        )
    ))
```

**Comment by carolinegoncalveszup** [comment]
Some suggestions for a cleaner style attributes: keep the display and positionType inside Style instead of inside Flex. The idea is that both positionType and display are not intrinsically related to flex.
Please let us know if you think it makes sense or not.

**Comment by diogooliveira-zup** [comment]
> 
> 
> Some suggestions for a cleaner style attributes: keep the display and positionType inside Style instead of inside Flex. The idea is that both positionType and display are not intrinsically related to flex.
> Please let us know if you think it makes sense or not.

We are doing this changes right now

File: android/beagle/src/main/java/br/com/zup/beagle/android/components/Spacer.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import android.view.View
+import br.com.zup.beagle.android.view.ViewFactory
+import br.com.zup.beagle.core.Style
+import br.com.zup.beagle.android.widget.RootView
+import br.com.zup.beagle.android.widget.WidgetView
+import br.com.zup.beagle.widget.core.Size
+import br.com.zup.beagle.widget.core.UnitType
+import br.com.zup.beagle.widget.core.UnitValue
+
+data class Spacer(
```
Comment (LG95) [inline android/beagle/src/main/java/br/com/zup/beagle/android/components/Spacer.kt:28]: According to #285 this component is gone.

File: android/beagle/src/test/java/br/com/zup/beagle/android/components/SpacerTest.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import br.com.zup.beagle.core.Style
+import br.com.zup.beagle.android.view.custom.BeagleFlexView
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+
+class SpacerTest : BaseComponentTest() {
```
Comment (uziasferreirazup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/components/SpacerTest.kt:29]: This call was deleted in the master

File: common/extended-schema/src/main/kotlin/br/com/zup/beagle/ext/WidgetExtensions.kt [hunk 6]
```diff
@@ -35,7 +35,7 @@ fun <T : Widget> T.setId(id: String) = this.apply { this.id = id }
  *
  * @return the current widget
  */
-fun <T : Widget> T.applyFlex(flex: Flex) = this.apply { this.flex = flex }
+fun <T : Widget> T.applyFlex(flex: Flex) = this.apply { style?.let { style?.copy(flex = flex) } ?: Style(flex = flex) }
```
Comment (LG95) [inline common/extended-schema/src/main/kotlin/br/com/zup/beagle/ext/WidgetExtensions.kt:38]:
```suggestion
fun <T : Widget> T.applyFlex(flex: Flex) = this.apply { this.style = (this.style ?: Style()).copy(flex = flex) }
```

**Pull request merged**